src/sentry/static/sentry/app/components/dropdownLink.jsx

```diff
@@ -1,98 +1,169 @@
     anchorRight: PropTypes.bool,
+    /** Keeps dropdown menu open when menu is clicked */
+    keepMenuOpen: PropTypes.bool,
     topLevelClasses: PropTypes.string,
     menuClasses: PropTypes.string
-  },
+  };
 
-  getDefaultProps() {
-    return {
-      disabled: false,
-      anchorRight: false,
-      caret: true
-    };
-  },
-  getInitialState() {
-    return {
+  static defaultProps = {
+    disabled: false,
+    anchorRight: false,
+    caret: true
+  };
+
+  constructor(...args) {
+    super(...args);
+    this.state = {
       isOpen: false
     };
-  },
-
-  componentDidMount() {
-    jQuery(this.refs.dropdownToggle).dropdown();
-    jQuery(this.refs.dropdownToggle.parentNode)
-      .on('shown.bs.dropdown', e => {
-        this.setState({
-          isOpen: true
-        });
-        this.props.onOpen && this.props.onOpen(e);
-      })
-      .on('hidden.bs.dropdown', e => {
-        setTimeout(() => {
-          if (!this.isMounted()) {
-            return;
-          }
-          this.setState({
-            isOpen: false
-          });
-          this.props.onClose && this.props.onClose(e);
-        });
-      });
-  },
+  }
 
   componentWillUnmount() {
-    jQuery(this.refs.dropdownToggle.parentNode).off();
-  },
+    $(document).off('click', this.checkClickOutside);
+  }
+
+  // Gets open state from props or local state when appropriate
+  isOpen = () => {
+    let {isOpen} = this.props;
+    let isControlled = typeof isOpen !== 'undefined';
+    return (isControlled && isOpen) || this.state.isOpen;
+  };
+
+  // Checks if click happens inside of dropdown menu (or its button)
+  // Closes dropdownmenu if it is "outside"
+  checkClickOutside = e => {
+    if (!this.dropdownMenu) return;
+    // Dropdown menu itself
+    if (this.dropdownMenu.contains(e.target)) return;
+    // Button that controls visibility of dropdown menu
+    if (this.dropdownActor.contains(e.target)) return;
+
+    this.handleClose(e);
+  };
+
+  // Opens dropdown menu
+  handleOpen = e => {
+    let {onOpen, isOpen} = this.props;
+    let isControlled = typeof isOpen !== 'undefined';
+    if (!isControlled) {
+      this.setState({
+        isOpen: true
+      });
+    }
+
+    if (typeof onOpen === 'function') {
+      onOpen(e);
+    }
+  };
+
+  // Closes dropdown menu
+  handleClose = e => {
+    let {onClose, isOpen} = this.props;
+    let isControlled = typeof isOpen !== 'undefined';
+    if (!isControlled) {
+      this.setState({isOpen: false});
+    }
+
+    if (typeof onClose === 'function') {
+      onClose(e);
+    }
+  };
+
+  // When dropdown menu is displayed and mounted to DOM,
+  // bind a click handler to `document` to listen for clicks outside of
+  // this component and close menu if so
+  handleMenuMount = ref => {
+    this.dropdownMenu = ref;
+
+    if (this.dropdownMenu) {
+      $(document).on('click', this.checkClickOutside);
+    } else {
+      $(document).off('click', this.checkClickOutside);
```

Comment:
is this how .off works? i think it takes a selector string

https://api.jquery.com/off/

Reply:
yeah: `.off( events [, selector ] [, handler ] )`